### ISSUE TITLE
Assertion Error when using new Java2DFrameConverter 

### ISSUE THREAD

**Issue by salamanders** [bytedeco/javacv#249]
I'm trying to play back a GoPro MP4 file, using the new Java2DFrameConverter to create a BufferedImage

```
    FFmpegFrameGrabber grabber = new FFmpegFrameGrabber(INPUT_VIDEO);
    Java2DFrameConverter converterToBI = new Java2DFrameConverter();

    grabber.start();
    final Frame frame0 = grabber.grab();
    Preconditions.checkNotNull(frame0);
    System.out.println(Java2DFrameConverter.getBufferedImageType(frame0));
    BufferedImage bi1 = converterToBI.convert(frame0);
```
throws a NPE or Assertion Error (when I enabled assertions)
```
Exception in thread "main" java.lang.AssertionError
	at org.bytedeco.javacv.Java2DFrameConverter.getBufferedImage(Java2DFrameConverter.java:609)
	at org.bytedeco.javacv.Java2DFrameConverter.getBufferedImage(Java2DFrameConverter.java:571)
	at org.bytedeco.javacv.Java2DFrameConverter.getBufferedImage(Java2DFrameConverter.java:568)
	at org.bytedeco.javacv.Java2DFrameConverter.convert(Java2DFrameConverter.java:66)
	at Merger.main(Merger.java:52)
```

It seems like a reasonable MP4 file:
```
Stream: GOPR0969.MP4
Path: ~/Pictures/GOPR0969.MP4

Duration: 0:17:43
Data Size: 3.72 GB
Bit Rate: 30.10 Mbps

Video Tracks:
GoPro AVC encoder, 1920 × 1440, 23.976 fps, 29.97 Mbps

Audio Tracks:
MPEG-4 Audio stereo, 48 kHz, 128 kbps

Stream Files:
GOPR0969.MP4 (3.72 GB)
```

**Comment by saudet** [comment]
You're using an old version of Java2DFrameConverter. Please try again with the latest version, and let me know if you still encounter that error. Thank you!

**Comment by salamanders** [comment]
I'm using maven 1.0 - will try with 1.0 download from here.

**Comment by saudet** [comment]
No, use 1.1- SNAPSHOT

**Comment by salamanders** [comment]
Unfortunately I'm new to Maven dev, so learning how to do that at the same time.  Do I need to add any other flags to make this work?
```
<dependency>
  <groupId>org.bytedeco</groupId>
  <artifactId>javacv</artifactId>
  <version>1.1-SNAPSHOT</version>
</dependency>
```

The POM for org.bytedeco:javacv:jar:1.1-SNAPSHOT is missing, no dependency information available

**Comment by saudet** [comment]
We need to build from source. Please follow the build instructions in the README.md file.

**Comment by salamanders** [comment]
https://github.com/bytedeco/javacpp was ok with a ```mvn install```, but building https://github.com/bytedeco/javacpp-presets seems to be a bigger deal.  Tried via ```cppbuild.sh -platform macosx-x86_64 install``` (not sure if this was fighting with my existing brew-install of opencv 2 and 3).  After a very big compile, ended with
```nasm: nasm: fatal: fatal: unrecognised output format `macho64' - use -hf for a listunrecognised output format `macho64' - use -hf for a list```

**Comment by saudet** [comment]
Yeah, some things want a newer nasm. Just brew install nasm, or just
cppbuild.sh install opencv if you're only interested with OpenCV.

**Comment by salamanders** [comment]
still no luck, may be easier to wait for the 1.1 release.  Was able to 'javacpp-presets/cppbuild.sh install opencv' but then the javacv/mvn install bails with ``` Failed to execute goal on project javacv: Could not resolve dependencies for project org.bytedeco:javacv:jar:1.1-SNAPSHOT: Could not find artifact org.bytedeco.javacpp-presets:flandmark:jar:1.07-1.1-SNAPSHOT```

**Comment by saudet** [comment]
You don't need to build flandmark if you're not interested by that.

Sure, you can wait for the next release, but if the bug still exists in the
next release, you'll have to wait for the release after that to get a fix...

**Comment by salamanders** [comment]
I'm all for trying!  What do I need to run instead of ```javacv/mvn install``` to just install the opencv and ffmpeg parts?

**Comment by saudet** [comment]
Just run 'cppbuild.sh install opencv ffmpeg'.

**Comment by salamanders** [comment]
I'm assuming you mean in javacpp-presets
Ran cppbuild.sh install opencv ffmpeg
which worked.

Not sure how to compile without all of them however, when I get to the last step
javacv>mvn package
I get ```[ERROR] Failed to execute goal on project javacv: Could not resolve dependencies for project org.bytedeco:javacv:jar:1.1-SNAPSHOT: Could not find artifact org.bytedeco.javacpp-presets:flandmark:jar:1.07-1.1-SNAPSHOT -> [Help 1]```

Which makes it sound like it expects all of them?

**Comment by salamanders** [comment]
Went into the pom.xml and commented out dependencies until it worked.

**Comment by salamanders** [comment]
Good news: Whohoo!  Got it to compile!
Bad news: 

```java.lang.AssertionError at org.bytedeco.javacv.Java2DFrameConverter.getBufferedImage(Java2DFrameConverter.java:614)```

**Comment by saudet** [comment]
Thanks for your effort! So, let's see, what does `Java2DFrameConverter.getBufferedImageType(frame0)` returns? And could you also print out all the fields of `frame0`? Thanks!

**Comment by dotd** [comment]
I have the same, or similar, bug. The return result of  Java2DFrameConverter.getBufferedImageType(frame0) is 5. What the different values mean?

**Comment by saudet** [comment]
@dotd Could you also print out all the fields of `frame0`? Thanks!

**Comment by salamanders** [comment]
That's odd.  2 audio no video?

```
  public static void main(final String[] args) throws Exception {
    FFmpegFrameGrabber g = new FFmpegFrameGrabber("/Users/ME/Pictures/Walkathon/GOPR0969.MP4");
    Java2DFrameConverter converter = new Java2DFrameConverter();
    g.start();
    for (int i = 0 ; i < 50 ; i++) {
      final Frame f = g.grab();
      System.out.println(f);;
      System.out.println(Java2DFrameConverter.getBufferedImageType(f));
      
      converter.convert(f);
      //ImageIO.write(g.grab(), "png", new File("frame-dump/video-frame-" + System.currentTimeMillis() + ".png"));
    }
    g.stop();
  }
```

```
Frame  (id=24)	
	audioChannels	2	
	image	null	
	imageChannels	0	
	imageDepth	0	
	imageHeight	0	
	imageStride	0	
	imageWidth	0	
	keyFrame	true	
	opaque	avutil$AVFrame  (id=27)	
		address	140213375726496	
		capacity	0	
		deallocator	null	
		limit	0	
		position	0	
	sampleRate	48000	
	samples	Buffer[2]  (id=31)	
		[0]	DirectFloatBufferU  (id=55)	
		[1]	DirectFloatBufferU  (id=59)	
			address	140213380369920	
			att	DirectByteBuffer  (id=61)	
			capacity	1024	
			hb	null	
			isReadOnly	false	
			limit	1024	
			mark	-1	
			offset	0	
			position	0	
```

**Comment by salamanders** [comment]
Whohoo!  final Frame f = g.grabImage(); works!  So it only dies when you use .grab().

**Comment by saudet** [comment]
Ah, yes, we need to check if there is an image. It should throw a clear
error message for that...
2015/10/19 1:01 "salamanders" <notifications@github.com>:

**Comment by saudet** [comment]
Fix included in JavaCV 1.1. Thanks for reporting!
